File: module-server/src/main/java/de/twomartens/wahlrecht/configuration/ClockConfiguration.java
package de.twomartens.wahlrecht.configuration;

import de.twomartens.wahlrecht.property.ServiceProperties;
import java.time.Clock;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@RequiredArgsConstructor
public class ClockConfiguration {

  private final ServiceProperties serviceProperties;

  @Bean
  public Clock clock() {
    return Clock.system(serviceProperties.getDefaultTimeZone());
  }

}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/configuration/ClockConfiguration.kt
package de.twomartens.wahlrecht.configuration

import de.twomartens.wahlrecht.property.ServiceProperties
import org.springframework.cloud.context.config.annotation.RefreshScope
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.Clock

@Configuration
open class ClockConfiguration(private val serviceProperties: ServiceProperties) {

    @Bean
    @RefreshScope
    open fun clock(): Clock {
        return Clock.system(serviceProperties.defaultTimeZone)
    }
}